### PR TITLE
Bundlebuilder patches, including generating AppStream data files!

### PR DESCRIPTION
**Pull request by samdroid-apps**
Probably replaces https://github.com/hughsie/appstream-glib/pull/111

### CODE REVIEW

**Comment by davelab6** [comment]
Nice!

File: src/sugar3/activity/bundlebuilder.py [hunk 1]
```diff
@@ -345,6 +411,61 @@ def _install_desktop_file(self, prefix, activity_path):
...
+            if not info.has_option('Activity', name):
+                print('[WARNING] Activity needs more metadata for AppStream '
+                      'file')
+                print('  Without an AppStream file, the activity will NOT '
+                      'show in software stores!')
+                print('  Please `pydoc sugar3.activity.bundlebuilder` for'
+                      'more info')
+                return
+
+        # See https://people.freedesktop.org/~hughsient/appdata/ for spec
```
Comment (ximion) [inline src/sugar3/activity/bundlebuilder.py]: Please don't use the old location, and rather the "new" spec:
https://www.freedesktop.org/software/appstream/docs/sect-Metadata-Application.html, which documents all valid tags excessively, or the quickstart guide: https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html#sect-Quickstart-DesktopApps

File: src/sugar3/activity/bundlebuilder.py [hunk 2]
```diff
@@ -345,6 +411,61 @@ def _install_desktop_file(self, prefix, activity_path):
...
+                ET.SubElement(e, 'image').text = screenshot
+
+        if info.has_option('Activity', 'url'):
+            ET.SubElement(root, 'url', type='homepage').text = \
+                info.get('Activity', 'url')
+        if info.has_option('Activity', 'repository_url'):
+            ET.SubElement(root, 'url', type='repository').text = \
+                info.get('Activity', 'repository_url')
+
+        path = os.path.join(prefix, 'share', 'appdata',
```
Comment (ximion) [inline src/sugar3/activity/bundlebuilder.py]: Please don't use the legacy path if possible (should be "/usr/share/metainfo/") - use the legacy path only if you need backwards compatibility with old AppStream tooling.

**Comment by samdroid-apps** [comment]
Thanks for the review @ximion!

File: src/sugar3/activity/bundlebuilder.py [hunk 3]
```diff
@@ -345,6 +411,61 @@ def _install_desktop_file(self, prefix, activity_path):
...
+                e = ET.SubElement(ss_root, 'screenshot')
+                if i == 0:
+                    e.set('type', 'default')
+                ET.SubElement(e, 'image').text = screenshot
+
+        if info.has_option('Activity', 'url'):
+            ET.SubElement(root, 'url', type='homepage').text = \
+                info.get('Activity', 'url')
+        if info.has_option('Activity', 'repository_url'):
+            ET.SubElement(root, 'url', type='repository').text = \
```
Comment (ximion) [inline src/sugar3/activity/bundlebuilder.py]: A repository URL type does not exist...
But you might want to add one for the bugtracker.
See https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-url for details on `<url/>`

**Pull request merged**